### PR TITLE
Added Win32 support for per-monitor DPI and scaling

### PR DESCRIPTION
**Pull request by stammen**
## Description
This PR adds Per Monitor DPI and Scaling support to the Win32 (Desktop and Centennial) versions. The PR follows the DPI aware guidance at:

https://msdn.microsoft.com/en-us/library/windows/desktop/dn469266(v=vs.85).aspx.  

Code was added to enable Kodi to resize the client and non-client portions of its window based on the DPI and scaling setting of the monitor on which the majority if its window is located. This PR was coded with guidance from the Display team at Microsoft.

## Motivation and Context
Kodi previously supported System wide dpi awareness. This allowed Kodi to correctly size its window at app startup. However, if the user moved the Kodi window to another monitor that has a different DPI or scaling setting, the window would not resize correctly to the new monitor settings. This PR adds the code needed to allow Kodi to handle monitors with different settings.

## How Has This Been Tested?
Note: This PR was tested by the Display team at Microsoft for correct functionality.

In order to correctly test this code, a multi-monitor setup is required.
1. Change the Scaling setting of one of the monitors using the Display Control Panel (see screenshot below).
2. Run Kodi and exit fullscreen display
3. Drag the Kodi window to the other monitor.
4. Windows 8.1 and higher will send Kodi a WM_DPICHANGED message with a recommended window size for the new display, Kodi will resize the window to this recommended size. The title bar and other non-client portions of the window will also be resized by the operating system based on the scaling or dpi of the new monitor location.

### Display Settings for 2nd Monitor
image

### Kodi on Monitor with 100% Scaling
image

### Kodi on Monitor with 150% Scaling
image

## Types of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Note: This code checks for the availability of the dpi aware apis and no changes will be made to Kodi running on Windows XP through Windows 7. Most of these changes will affect Kodi running on Windows 8.1 and Windows 10.

## Checklist:
- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
thx
@Paxxi @afedchin ping
jenkins build this please

**Review by afedchin — APPROVED** [review]
looks good. thanks.

File: xbmc/windowing/WinSystem.h [hunk 1]
```diff
@@ -65,6 +65,7 @@ class CWinSystemBase
   virtual bool DestroyWindow(){ return false; }
   virtual bool ResizeWindow(int newWidth, int newHeight, int newLeft, int newTop) = 0;
   virtual bool SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, bool blankOtherDisplays) = 0;
+  virtual bool DPIChanged(WORD dpi, RECT windowRect) = 0;
```
Comment (afedchin) [inline xbmc/windowing/WinSystem.h]: please drop this declaration it breaks build on other platforms. the declaration in WinSystemWin32.h is enough

**Review by Paxxi — APPROVED** [review]
Looks good to me, thank you!

File: xbmc/windowing/WinSystem.h (hunk 1 above)
Comment (stammen) [inline xbmc/windowing/WinSystem.h]: Fixed as suggested. Thanks!

**Comment by afedchin** [comment]
jenkins build this please

**Pull request merged**